modules/xp/index.ts

```diff
@@ -78,55 +78,7 @@ defineSubcommands(
 				});
 			}
 			case "top": {
-				const top = [...xpDatabase.data].sort((one, two) => two.xp - one.xp);
-
-				const { user } = options.options;
-				const index = user ? top.findIndex(({ user: id }) => id === user.id) : undefined;
-				if (index === -1) {
-					return await interaction.reply({
-						content: `${
-							constants.emojis.statuses.no
-						} ${user?.toString()} could not be found! Do they have any XP?`,
-
-						ephemeral: true,
-					});
-				}
-
-				await paginate(
-					top,
-					async (xp) =>
-						`**Level ${getLevelForXp(Math.abs(xp.xp)) * Math.sign(xp.xp)}** - ${
-							getSettings(interaction.user).useMentions
-								? `<@${xp.user}>`
-								: (
-										await client.users
-											.fetch(xp.user)
-											.catch(() => ({ displayName: `<@${xp.user}>` }))
-								  ).displayName
-						} (${Math.floor(xp.xp).toLocaleString("en-us")} XP)`,
-					(data) => interaction.reply(data),
-					{
-						title: "XP Leaderboard",
-						singular: "user",
-
-						user: interaction.user,
-						rawOffset: index,
-
-						generateComponents() {
-							return getSettings(interaction.user, false).useMentions === undefined
-								? [
-										{
-											customId: "levelUpPings_toggleSetting",
-											type: ComponentType.Button,
-											label: "Toggle Mentions",
-											style: ButtonStyle.Success,
-										},
-								  ]
-								: undefined;
-						},
-						customComponentLocation: "below",
-					},
-				);
+				await XPLeaderboard(interaction, options.options.user as User);
```

Comment:
Don't do `as User`, the function already supports GuildMembers, just change the signature

Reply:
For some reason TypeScript was screaming at me when I didn't do `as User`

Reply:
Oh wait now I remember. I just have to add `.user`

Reply:
no, it was because you hadn't changed the function signature